src/humancellatlas/data/metadata/helpers/staging_area.py

```diff
@@ -219,117 +207,103 @@ def _entity_ids_by_type(self,
+    def _get_subtree(cls, root: git.Tree, path: tuple[str, ...]) -> git.Tree:
+        while len(path) > 0:
+            p, *path = path
+            for tree in root.trees:
+                if tree.name == p:
+                    root = tree
+                    break
+            else:
+                raise ValueError(f'Could not find path {p!r} in {root.path!r}')
+        return root
 
     def load_staging_area(self) -> StagingArea:
-        staging_area_folders = self._get_folders(path=self.path)
+        staging_area_folders = {t.name for t in self.tree.trees}
         expected_folders = {'data', 'descriptors', 'links', 'metadata'}
         require(set(staging_area_folders) == expected_folders,
-                f'{self.repo.full_name} {self.path} is not a valid staging area')
-        links = self._get_files(path=self.path + ('links',))
-        metadata = {}
-        for folder in self._get_folders(path=self.path + ('metadata',)):
-            files = self._get_files(path=self.path + ('metadata', folder))
-            metadata.update(files)
-        descriptors = {}
-        for folder in self._get_folders(path=self.path + ('descriptors',)):
-            files = self._get_files(path=self.path + ('descriptors', folder))
-            descriptors.update(files)
+                f'{self.url} {self.tree.path} is not a valid staging area')
+        links, metadata, descriptors = self._get_files()
         return StagingArea(links=links, metadata=metadata, descriptors=descriptors)
 
-    def _get_folders(self, path: Tuple[str, ...]) -> List[str]:
-        folders = []
-        log.debug('Getting list of folders in %s', path)
-        try:
-            path_contents = self.repo.get_contents(path='/'.join(path), ref=self.ref)
-        except UnknownObjectException as e:
-            raise ValueError('Github path not found', self.repo.full_name, self.path) from e
-        for content in path_contents:
-            if content.type == 'dir':
-                folders.append(content.name)
-        return folders
+    def _parse_blob_path(self, blob: git.Blob) -> tuple[str, str]:
+        parts = blob.path.split('/')
+        for i, name in enumerate(self.path):
+            assert name == parts[i], name
+        folder = parts[len(self.path)]
+        file_name = parts[-1]
+        return folder, file_name
+
+    def _get_files(self) -> tuple[
+        dict[str, LinksFile],
+        dict[str, MetadataFile],
+        dict[str, DescriptorFile]
+    ]:
+        links = {}
+        metadata = {}
+        descriptors = {}
+        for item in self.tree.traverse():
+            if item.type == 'blob' and item.path.endswith('.json'):
+                log.debug('Reading file %s', item.path)
+                folder, file_name = self._parse_blob_path(item)
+                file_contents = self.repo.git.show(f'{self.commit.hexsha}:{item.path}')
+                file_json = json.loads(file_contents)
```

Comment:
I see a type warning on this line that indicates that `json` is being resolved to `src/humancellatlas/data/metadata/helpers/json.py` instead of the standard python module. Is that just a PyCharm bug, or is there an actual problem with the incorrect module being loaded?

Possibly related: https://github.com/DataBiosphere/azul/issues/3133

Reply:
Renamed humancellatlas's `json.py` to avoid the conflict

Reply:
There must be a less invasive way to deal with the type warning.